metrics-recorder-text/src/lib.rs

```diff
@@ -192,47 +210,47 @@ impl MetricsTree {
     }
 }
 
-impl From<TextRecorder> for String {
-    fn from(r: TextRecorder) -> String {
-        let mut structure = r.structure;
-        for (key, h) in r.histos {
-            let (name_parts, name) = key_to_parts(key);
-            let mut values = hist_to_values(name, h, &r.quantiles);
+impl Export<String> for TextRecorder {
+    fn export(&self) -> String {
+        let mut structure = self.structure.clone();
+        for (key, h) in &self.histos {
+            let (name_parts, name) = key_to_parts(key.clone());
+            let mut values = hist_to_values(name, h.clone(), &self.quantiles);
             structure.insert(name_parts, &mut values);
         }
-        structure.into_output()
+        structure.render()
     }
 }
 
-enum SortEntry {
-    Inline(String),
-    Nested(String, MetricsTree),
+enum SortEntry<'a> {
+    Inline(&'a String),
+    Nested(&'a String, &'a MetricsTree),
 }
 
-impl SortEntry {
-    fn name(&self) -> &String {
+impl<'a> SortEntry<'a> {
+    fn name(&self) -> &'a String {
         match self {
             SortEntry::Inline(s) => s,
             SortEntry::Nested(s, _) => s,
         }
     }
 }
 
-impl PartialEq for SortEntry {
+impl<'a> PartialEq for SortEntry<'a> {
```

Comment:
I think you can elide this lifetime `SortEntry<'_>`?

Reply:
damn i didn't know this was a thing. cool.